### DISCUSSION TITLE
Trying to generate tailwind css with postcss using server endpoint but getting error missing preflight.css when building

### DISCUSSION THREAD

**Discussion by andyjamesn** [nuxt/nuxt#27432] [Questions]
I have a basic fresh Nuxt install and 1 server api file. The goal is to generate a tailwind css file programmatically using postcss.

It works fine in dev mode but as soon as I deploy it fails with the error:

No such file or directory `.output/server/node_modules/tailwindcss/lib/css/preflight.css`

![CleanShot 2024-06-03 at 03 12 21@2x](https://github.com/nuxt/nuxt/assets/54320410/2081879d-0e80-4fe0-a723-9ef636098131)

When I look in `.output/server/node_modules/tailwindcss//lib/css/preflight.css` I do not see the css folder or the preflight.css file

In my main node_modules I can see the /lib/css/preflight.css folder and file. 

The normal node_modules/tailwindcss in my project root:
![CleanShot 2024-06-03 at 03 33 08@2x](https://github.com/nuxt/nuxt/assets/54320410/61ef2c42-078a-4f10-a534-0f9568eeb2d4)

The node_modules/tailwindcss in output is quite different. It looks like it excludes the cli code and the preflight. 
![CleanShot 2024-06-03 at 03 34 18@2x](https://github.com/nuxt/nuxt/assets/54320410/0b6f1162-5887-4bd7-8122-9138f169b0e6)

Is Nuxt is doing something to tailwind at build time to strip down the node module? Is there a way to prevent this?

I have a codesandbox of the code but since it is a build issue I am not sure it helps:

https://codesandbox.io/p/devbox/exciting-lumiere-xcy5mm

But here is minimal version of the server code that generates the css. 

```javascript
import tailwindcss from "tailwindcss";
import tailwindConfig from "@/tailwind.config.js";
import postcss from "postcss";

const withConfigPlugin = (config = {}) => {
  return postcss([tailwindcss(config)]);
};

async function generateCss(cssInput) {
  const postCssConfig = [withConfigPlugin(tailwindConfig)];
  let css = "";
  return postcss(postCssConfig)
    .process(cssInput, { from: undefined })
    .then((result) => {
      css = result.css;
      return css;
    });
}

export default defineEventHandler(async (event) => {
  const css = await generateCss(`@tailwind base;
  @tailwind components;
  @tailwind utilities;body { @apply bg-red-500;}`);
  // console.log('css', css);
  return {
    statusCode: 200,
    body: {
      css
    },
  };
});
```

**Comment by mirite** [comment]
Oddly enough, I experienced the same issue when upgrading to Next.js 15. It seems to remove the css files in the server bundle that were previously included.

I managed to fix by disabling the preflight in the tailwind config (if you don't need the preflight at runtime).
```javascript
function generateCss(str) {
  const config = {
    plugins: [],
    content: [],
    corePlugins: {
      preflight: false,
    },
  }
  
  const processed = await postcss([
      tailwindcss(config),
      autoprefixer,
    ]).process(str, { from: undefined })
  return process.css
}
```
